### ISSUE TITLE
'isComplete` in numeric extensions doesn't take into account 'negationSymbol'

### ISSUE THREAD

**Issue by DmitrySenin** [RobinHerbots/Inputmask#1844]
The current implementation doesn't remove characters of 'negationSymbol':
```javaScript
isComplete: function (buffer, opts) {
    var maskedValue = buffer.join(""),
        bufClone = buffer.slice();
    //verify separator positions
    if (bufClone.join("") !== maskedValue) return false;

    var processValue = maskedValue.replace(opts.prefix, "");
    processValue = processValue.replace(opts.suffix, "");
    processValue = processValue.replace(new RegExp(Inputmask.escapeRegex(opts.groupSeparator) + "([0-9]{3})", "g"), "$1");
    if (opts.radixPoint === ",") processValue = processValue.replace(Inputmask.escapeRegex(opts.radixPoint), ".");
    return isFinite(processValue);
}
```
so the following settings will always lead to clearing of the input:
```javaScript
Inputmask('numeric', {
            negationSymbol: {
                front: "(",
                back: ")"
            },
            clearIncomplete: true
    })
```

**Comment by RobinHerbots** [comment]
@DmitrySenin , 

This should do it, can you verify & confirm.

```
  isComplete: function (buffer, opts) {
                var maskedValue = buffer.join(""),
                    bufClone = buffer.slice();
                //verify separator positions
                if (bufClone.join("") !== maskedValue) return false;

                 var processValue = maskedValue.replace(new RegExp("^" + Inputmask.escapeRegex(opts.negationSymbol.front)), "-");
                processValue = processValue.replace(new RegExp(Inputmask.escapeRegex(opts.negationSymbol.back) + "$"), "");
                processValue = processValue.replace(opts.prefix, "");
                processValue = processValue.replace(opts.suffix, "");
                processValue = processValue.replace(new RegExp(Inputmask.escapeRegex(opts.groupSeparator) + "([0-9]{3})", "g"), "$1");
                if (opts.radixPoint === ",") processValue = processValue.replace(Inputmask.escapeRegex(opts.radixPoint), ".");
                return isFinite(processValue);
            },
```

**Comment by DmitrySenin** [comment]
@RobinHerbots ,
Thanks for such a fast response.
I can confirm that the new version of the method did the trick, I had a test locally which found the problem and it passed successfully:
```javascript

    qunit.test("numeric + (negationSymbol = parentheses) + (clearIncomplete = true) + type -123. then blur", function (assert) {
        var $fixture = $("#qunit-fixture");
        $fixture.append('<input type="text" id="testmask" />');
        var testmask = document.getElementById("testmask");
        Inputmask("numeric", {
            rightAlign: false,
            autoGroup:  true,
            groupSize: 3,
            digits: 3,
            enforceDigitsOnBlur: true,
            groupSeparator: ',',
            negationSymbol: {
                front: "(",
                back: ")"
            },
            clearIncomplete: true
        }).mask(testmask);
        testmask.focus();
        $("#testmask").Type("-123.");
        testmask.blur();

        assert.equal(testmask.value, "(123.000)", "Result " + testmask.value);
    });
```
I also played with UI and the input is not cleared upon focus out neither for parentheses nor for standard formatting(with minus)

Thanks

**Comment by RobinHerbots** [comment]
@DmitrySenin , 

If you have more unit tests.  Please contribute them.  They are always usefull!

What do you mean exactly with  "not cleared upon focus out neither for parentheses nor for standard formatting(with minus)"

**Comment by DmitrySenin** [comment]
@RobinHerbots ,
I'm trying to fix #1836 and am going to add new unit-tests in scope of it.

Under "not cleared upon focus out neither for parentheses nor for standard formatting(with minus)", I meant that I tested not only with the problematic configuration:
```javascript
Inputmask('numeric', {
            negationSymbol: {
                front: "(",
                back: ")"
            },
            clearIncomplete: true
    })
```
but also with the standard representation of numbers - the configuration below(I think it's the default one):
```javascript
Inputmask('numeric', {
            negationSymbol: {
                front: "-",
                back: ""
            },
            clearIncomplete: true
    })
```

**Comment by RobinHerbots** [comment]
If you submit the unittest, they start to annoy me when I run tests and
they will get my attention sooner ;-)

Op di 20 mrt. 2018 om 12:17 schreef Dmitry Senin <notifications@github.com>:
